src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -467,15 +470,32 @@ private void RenderUnboundAttribute(string attributeName, Chunk attributeValueCh
                     // Dynamic attribute value should be run through the conditional attribute removal system. It's
                     // unbound and contains C#.
 
+                    // TagHelper attribute rendering is buffered by default. We do not want to write to the current
+                    // writer.
+                    var currentTargetWriter = _context.TargetWriterName;
```

Comment:
Move this down to where it has effect where we do the `_attributeCodeVisitor.Accept(attributeValueChunk);`